cmds/xkcd.cmd.js

```diff
@@ -18,101 +16,51 @@ const formatReply = function(comic) {
 	};
 };
 
-const fetchComic = function(id, done) {
-	if(typeof id === 'function' && typeof done === 'undefined') {
-		done = id;
-		id = null;
-	} else {
+const fetchComic = (id) => {
+	return new Promise((resolve, reject) => {
 		try {
-			id = Number(id);
+			if (!id) {
+				return xkcd(resolve);
+			}
+			return xkcd(_.toInteger(id), resolve);
 		} catch (err) {
-			return done(err);
+			return reject(err);
 		}
-	}
-
-	try {
-		if(id === null) {
-			return xkcd(function(data) {
-				return done(null, data);
-			});
-		}
-
-		return xkcd(id, function(data) {
-			return done(null, data);
-		});
-	} catch (err) {
-		return done(err);
-	}
+	});
 };
 
 module.exports = {
 	url    : '/xkcd',
-	handle : function(msg, info, reply, error) {
-		// If user did not supply any params, they want the latest comic.
-		if(!msg) {
-			return fetchComic(function(err, comic) {
-				if(err) {
-					err.message += ` ${JSON.stringify(matches)}`;
-					return error(err);
-				}
+	handle : async (msg, info, reply, error) => {
+		try {
+			// If user did not supply any params, they just want the latest comic.
+			if (!msg) {
+				const comic = await fetchComic();
 				return reply(formatReply(comic));
-			});
-		}
+			}
 
-		// If user entered `/xkcd #<num>`, they're requesting a specific comic by ID.
-		var matches = msg.match(/^[\s]*#([\d]+)[\s]*$/);
-		if(matches !== null && matches.length > 1 && matches[1]) {
-			return fetchComic(matches[1], function(err, comic) {
-				if(err) {
-					err.message += ` ${JSON.stringify(matches)}`;
-					return error(err);
-				}
+			// If user entered `/xkcd #<num>`, they're requesting a specific comic by ID.
+			const matches = msg.match(/^\s*#(\d+)\s*$/);
+			if (_.has(matches, 1)) {
```

Comment:
Interesting way to do this check!

Reply:
This is the first time ever that I've done it this way and I'm really liking it.  I'm so sick of lengthy if-statements after regex matching that are like `if (matches !== null && matches.length > 1 && matches[1])`